packages/web-react/src/hooks/useCancelEvent.ts

```diff
@@ -1,25 +1,61 @@
-import { useCallback, useEffect, MutableRefObject } from 'react';
+import { useCallback, MutableRefObject } from 'react';
+import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect';
+
+// TODO: Remove `isOpen` and listeners with `handleKeyDown` when Chrome fixes the bug,
+// right now Chrome is bugged and sends un-cancelable events, so closing modal based on
+// `cancel` event is not possible in chrome.
+// FireFox and Safari are working fine.
 
 const EVENT_CANCEL = 'cancel';
+const EVENT_KEYDOWN = 'keydown'; // ! Removed when bug in chrome will be fixed
 
-export const useCancelEvent = (ref: MutableRefObject<HTMLElement | null>, callback: (event: Event) => void) => {
+export const useCancelEvent = (
+  ref: MutableRefObject<HTMLElement | null>,
+  callback: (event: Event) => void,
+  closeOnEscapeKeyDown: boolean = true,
+  isOpen: boolean = false, // ! Removed when bug in chrome will be fixed
+) => {
   const handleCancel = useCallback(
     (event: Event) => {
+      // Do nothing if there is no reference or no callback
+      if (!ref || !callback) {
+        return;
+      }
+
+      // Do nothing if the event was already processed.
+      if (event.defaultPrevented) {
+        return;
+      }
+
       event.preventDefault();
 
-      callback(event);
+      if (callback && closeOnEscapeKeyDown) {
+        callback(event);
+      }
+    },
+    [ref, callback, closeOnEscapeKeyDown],
+  );
+
+  // ! Removed when bug in chrome will be fixed
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && !closeOnEscapeKeyDown && isOpen) {
```

Comment:
```suggestion
      if (event.key === KEY_ESCAPE && !closeOnEscapeKeyDown && isOpen) {
```

Please, set a constant for this :-)

Reply:
constant created